### PR TITLE
Action job for building macOS corplink-rs

### PR DESCRIPTION
**Pull request by overvenus**
Add an action jobs for building macOS corplink-rs on release.

Examples:
* Action: https://github.com/overvenus/corplink-rs/actions/runs/9394276177
* Release: https://github.com/overvenus/corplink-rs/releases/tag/4.3

### CODE REVIEW

**Comment by overvenus** [comment]
It's not ready yet, will contribute later.

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "nightly-2024-06-01"
```
Comment (PinkD) [inline rust-toolchain.toml:2]: 这个不是必须的？

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -7,24 +7,30 @@ on:
...
           tag_name: ${{ github.ref }}
           release_name: ${{ github.ref }}
         env:
-          # (required)
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build-arch-linux:
+    name: build-arch-linux
+    needs: ['create-release']
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (overvenus) [reply .github/workflows/release.yml:24]: 不清楚，这块我没改动，之前也是 `runs-on: ubuntu-latest`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -7,24 +7,30 @@ on:
...
         uses: actions/create-release@v1
         with:
           tag_name: ${{ github.ref }}
           release_name: ${{ github.ref }}
         env:
-          # (required)
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build-arch-linux:
+    name: build-arch-linux
```
Comment (overvenus) [inline .github/workflows/release.yml:22]:
```suggestion
    name: build-archlinux
```

**Review by PinkD — APPROVED** [review]

**Pull request merged**